onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc

```diff
@@ -95,7 +95,8 @@ Status AddInitializerInNewLayout(ModelBuilder& model_builder,
                                  bool is_conv) {
   const auto& tensor = *model_builder.GetInitializerTensors().at(name);
   auto data_type = tensor.data_type();
-  if (data_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+  if (data_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT16 &&
+      data_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
```

Comment:
May be worth pulling this check into a little shared helper, like `IsSupportedDataType()`...